stacks_queues/queue.py

```
@@ -15,47 +15,86 @@ def __init__(self):
         self.front = -1
         self.rear = -1
         self.size = 0
-      
+
 
     def enqueue(self, element):
         """ Adds an element to the Queue
             Raises a QueueFullException if all elements
             In the store are occupied
             returns None
         """
-        pass
+        if self.size == 0:
+            self.front = 0
+
+        elif self.size == self.buffer_size:
+            raise QueueFullException("Error message")
+
+        self.rear = (self.rear + 1) % self.buffer_size
+        self.store[self.rear] = element
+        self.size += 1
+        return None
 
     def dequeue(self):
         """ Removes and returns an element from the Queue
             Raises a QueueEmptyException if 
             The Queue is empty.
         """
-        pass
+        try:
+            temp = self.store[self.front]
+            self.store[self.front] = None
+            self.front = (self.front + 1) % self.buffer_size
+            self.size -= 1
+            return temp
+        except:
+            raise QueueEmptyException("Error message")
 
     def front(self):
         """ Returns an element from the front
             of the Queue and None if the Queue
             is empty.  Does not remove anything.
         """
-        pass
-        
+        try:
+            return self.store[self.front]
+        except:
+            raise QueueEmptyException("Error message")
 
     def size(self):
         """ Returns the number of elements in
             The Queue
         """
-        pass
+        try:
+            return self.size
+        except:
+            raise QueueEmptyException("Error message")
+        
 
     def empty(self):
```

👍   I don't think you need to raise an exception here.